composer.json

```diff
@@ -21,7 +21,6 @@
         "paragonie/sodium_compat": "^1.13",
         "symfony/validator": "^4.4 || ^5 || ^6",
         "myclabs/deep-copy": "^1.10.2",
-        "guzzlehttp/promises": "^1.5",
```

Comment:
In discussion with @tedbow, we agreed that we don't really need the potential asynchronicity offered by promises. So this PR removes promises support, and therefore we don't need this dependency anymore.